pinot-query-runtime/src/main/java/org/apache/pinot/query/runtime/operator/IntersectOperator.java

```diff
@@ -58,6 +58,7 @@ public String toExplainString() {
 
   @Override
   protected boolean handleRowMatched(Object[] row) {
-    return _rightRowSet.remove(new Record(row));
+    Record record = new Record(row);
+    return _rightRowSet.remove(record, _rightRowSet.count(record)) != 0;
```

Comment:
Can we use `MultiSet.setCount` here instead? That should mean to execute one less lookup.